llama_stack/distribution/stack.py

```diff
@@ -172,7 +172,6 @@ def replace_env_vars(config: Any, path: str = "") -> Any:
                             # Create a copy with resolved provider_id but original config
                             disabled_provider = v.copy()
                             disabled_provider["provider_id"] = resolved_provider_id
-                            result.append(disabled_provider)
```

Comment:
if you do this, why do you need to remove the disabled providers then? i.e., you don't need any other changes right?